src/core/common/logging.hpp

```diff
@@ -1500,6 +1500,31 @@ const char *otLogRegionToString(otLogRegion aRegion);
 #define _otPlatLog(aLogLevel, aRegion, aFormat, ...)                        \
     OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION(aLogLevel, aRegion, aFormat, ## __VA_ARGS__)
 
+
+/*
+ * This provides a prototype for the platform defined logging function
+ *
+ * It is not intended to be a Doxygen comment because ...
+ * it is a macro that is used to override the normal function.
+ *
+ * The macro "OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION" can be defined
+ * by three methods
+ *
+ * Method (A) via default 'openthread-core-default-config.h' as described above.
+ *
+ * Method (B) a platform specific override header that predefined the macro
+ *
+ * Method (C) on the compile command line, via a "-D" type define.
+ *
+ * In the "-D" case, there might not be a protytpe yet
+ * this just provides that missing prototype in the form required
+ *
```

Comment:
I think we only need the two types `otLogRegion` and `otLogLevel` which are defined in a separate public header file "logging.h".... I think we can include this header file from the "OPENTHREAD_PROJECT_CORE_CONFIG_FILE" (or even better from the header file declaring the function replacing the `otPlatLog`)...

Here is how we are using it internally:
We have a project specific config file which has
````
/**
 * @def OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION
 *
 * Defines the name of function/macro used for logging inside OpenThread, by default is set to `otPlatLog()`.
 *
 */
#include <header file where function/macro MY_LOGGER is declared/defined>
#define OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION     MY_LOGGER
...
````
And in header file where `MY_LOGGER` is defined/declraed we include `"openthread/platform/logging.h"` to get the types.





The whole `OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION` model was added so we can declare the platform logging as a macro instead of a function `otPlatLog`... (which was a requirement on an internal platform). When we have a function then I'd think better approach to define the `otPlatLog()` function directly (similar to the way `OPENTHREAD_CONFIG_ENABLE_DEFAULT_LOG_OUTPUT` maps it to some internal function in either cli or ncp). Would this work?